boot/freeldr/freeldr/arch/uefi/amd64/uefiasm.S

```diff
@@ -46,6 +45,12 @@ _exituefi:
     pop rsi
     pop rbp
 
+#ifdef _USE_ML
+    lgdt fword ptr [_gdtptr]
+#else
+    lgdt cs:[_gdtptr][rip] /* GAS isn't my friend - avoid letting it generate absolute addressing */
```

Comment:
Nice that you've found how to fix that error. Will be helpful in other cases as well.